circuit-overlay/src/commonMain/kotlin/com/slack/circuit/overlay/ContentWithOverlays.kt

```diff
@@ -18,17 +24,48 @@ import androidx.compose.ui.Modifier
  * @param overlayHost the [OverlayHost] to use for managing overlays.
  * @param content The regular content to render. Any overlays will be rendered over them.
  */
+@OptIn(ExperimentalAnimationApi::class)
 @Composable
 public fun ContentWithOverlays(
   modifier: Modifier = Modifier,
   overlayHost: OverlayHost = rememberOverlayHost(),
   content: @Composable () -> Unit
 ) {
   CompositionLocalProvider(LocalOverlayHost provides overlayHost) {
-    val overlayHostData by rememberUpdatedState(overlayHost.currentOverlayData)
     Box(modifier) {
       content()
-      key(overlayHostData) { overlayHostData?.let { data -> data.overlay.Content(data::finish) } }
+      val transition = updateTransition(overlayHost.currentOverlayData, label = null)
+      val enterTransition =
+        remember(transition.targetState) {
+          (transition.targetState?.overlay as? AnimatedOverlay)?.enterTransition
+            ?: EnterTransition.None
+        }
+      val exitTransition =
+        remember(transition.currentState) {
+          (transition.currentState?.overlay as? AnimatedOverlay)?.exitTransition
+            ?: ExitTransition.None
+        }
+      transition.AnimatedVisibility(
+        visible = { it != null },
+        modifier = Modifier.fillMaxSize(),
```

Comment:
Hmmm, this `fillMaxSize()` will affect the `content` layout too. I think it's safer to remove this and let the overlay content define what it wants to do.